kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -306,19 +326,99 @@ private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaClust
+                .map(VirtualKafkaClusterSpec::getTargetCluster)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster::getTls)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.targetcluster.Tls::getCertificateRef)
+                .map(certRef -> {
+                    return new Tls(new KeyPair(
+                            virtualClusterTargetTlsKeyPath(context, cluster, certRef).containerPath().toString(),
+                            virtualClusterTargetTlsCertPath(context, cluster, certRef).containerPath().toString(),
+                            null),
+                            null, null, null);
+                });
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsKeyPath(Context<KafkaProxy> context,
+                                                                         VirtualKafkaCluster cluster,
+                                                                         CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.key");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsCertPath(Context<KafkaProxy> context,
+                                                                          VirtualKafkaCluster cluster,
+                                                                          CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.crt");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsPath(Context<KafkaProxy> context,
+                                                                      VirtualKafkaCluster cluster,
+                                                                      CertificateRef certRef,
+                                                                      String file) {
+        if ("Secret".equals(certRef.getKind())
+                && certRef.getGroup().isEmpty()) {
+            // Assume it's compatible with a Secret of `type: kubernetes.io/tls`
+            String other = "vc-target-tls-" + ResourcesUtil.name(cluster);
+            Path volumeMountPath = Path.of("/opt/kroxylicious/cluster")
+                    .resolve(ResourcesUtil.name(cluster))
+                    .resolve("target")
+                    .resolve("tls-key");
+            Volume volume = new VolumeBuilder()
+                    .withName(other)
+                    .withNewSecret()
+                    .withSecretName(certRef.getName())
+                    .addNewItem("tls.key", null, "tls.key")
+                    .addNewItem("tls.crt", null, "tls.crt")
+                    .endSecret()
+                    .build();
+            VolumeMount volumeMount = new VolumeMountBuilder()
+                    .withName(other)
+                    .withMountPath(volumeMountPath.toString())
+                    .withReadOnly(true)
```

Comment:
I'm not using subpath mounts, which means the kubelet will refresh the files in the container if the Secret gets updated. This could be a way to support dynamic cert update in the future, but _something_ needs to care about certificate updates:

* One way would be for the proxy to have some background thread watching/checking the certificate periodically, have it terminate, and let the deployment re-create the pod. 

* Another way is to have the operator watch the Secret and delete pods when it changes.

Do we see this as a problem we want to solve right now, or do we want to say it's a user responsibility to intervene when their certs get reissued?